app/views/compute_resources/form/_azurerm.html.erb

```diff
@@ -3,8 +3,6 @@
 <%= text_f f, :user, :label => _('Subscription ID'), :required => true %>
 <%= text_f f, :uuid, :label => _('Tenant ID'), :required => true %>
 
-<%= selectable_f(f, :url, f.object.class.regions, {}, {:label => _('Azure Region'), :label_help => _("Some of the listed regions might not be supported for your subscription. Please verify on Azure portal."), :required => true }) %>
+<% regions = f.object.regions rescue [] %>
```

Comment:
Why do you want to ignore exceptions here? You will probably want to show it in the UI as message.
It would be better to move this sort of logic into a helper including the view part.

Reply:
I referred to how [GCE](https://github.com/theforeman/foreman/blob/34876ef54276dfd069cc9d37bdc35bbc7b735f5e/app/views/compute_resources/form/_gce.html.erb#L5) and [EC2](https://github.com/theforeman/foreman/blob/a6cf3a531b524d85c94fb0acefa3bb72b5e7574a/app/views/compute_resources/form/_ec2.html.erb#L8) handled the exception.
But, yeah it does make sense to define such logic to a helper.

Reply:
This select field is available on compute resource form on load. 
On `Load Regions` click it populates the regions list because before that user needs to fill other fields that are required for successful connection. 
Till this point `client.regions` might fail with connection failures or exception for required fields like Client ID.
Yes, might be useful to have helper but not very sure.